### ISSUE TITLE
Question about AdamW optimizer implementation in NSM

### ISSUE THREAD

**Issue by RosettaWYzhang** [sebastianstarke/AI4Animation#82]
Hi! Thanks for providing the code. 

For Neural State Machine, in the paper you mentioned that you are using AdamWR optimizer. However, in the code (https://github.com/sebastianstarke/AI4Animation/blob/master/AI4Animation/SIGGRAPH_Asia_2019/TensorFlow/NSM/Lib_Optimizer/AdamW.py), why is weight decay (wdc) is only used in _apply_sparse and not in _apply_dense? Is this optimizer just vanilla Adam optimizer?
